lib/api/v1/controllers/auth_test.go

```diff
@@ -0,0 +1,78 @@
+package controllers_test
+
+import (
+	"net/http"
+
+	"github.com/markgravity/golang-ic/lib/api/v1/controllers"
+	"github.com/markgravity/golang-ic/lib/api/v1/forms"
+	"github.com/markgravity/golang-ic/test"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AuthController", func() {
+	Describe("POST /sign-up", func() {
+		AfterEach(func() {
+			test.CleanUpDatabase()
+		})
+
+		Context("Given VALID payload", func() {
+			It("returns status OK", func() {
+				payload := map[string]interface{}{
+					"email":                 "test@gmail.com",
+					"password":              "test123",
+					"password_confirmation": "test123",
+				}
+
+				ctx, resp := test.MakeRequest(http.MethodPost, "/auth/sign-up", nil, payload)
+				controller := controllers.AuthController{}
+
+				controller.SignUp(ctx)
+
+				Expect(resp.Code).To(Equal(http.StatusOK))
+			})
+		})
+
+		Context("Given INVALID payload", func() {
+			It("returns an unprocessable entity status", func() {
```

Comment:
```suggestion
			It("returns the unprocessable entity status", func() {
```

As a quick rule of thumb:
- use "a / an" when refering to an unknown item
  e.g. 1: there are many chair. I need to seat on a chair (here `a` means `any chair will do`, we don't care of the identity of the chair)
  e.g. 2: there are many chaer. I need to seat on the blue chair (here `the` means `their is only 1 bule chair, I can identity it)

As there is only 1 unprocessable entity status (422), we return `the` status.
In the case there would be multiple statuses, we would use `a`. For example "it returns a 40{X} status" (meaning any of the 400, 401, 402, 403, ... statuses).